src/sinks/rabbitmq.rs

```diff
@@ -0,0 +1,148 @@
+use crate::{
+  buffers::Acker,
+  event::{self, Event},
+  sinks::util::MetadataFuture,
+  topology::config::{DataType, SinkConfig},
+  Error,
+};
+use futures::{
+  future::{self, poll_fn, IntoFuture},
+  stream::FuturesUnordered,
+  Async, AsyncSink, Future, Poll, Sink, StartSend, Stream,
+};
+use lapin::options::{BasicPublishOptions, QueueDeclareOptions};
+use lapin::types::FieldTable;
+use lapin::{BasicProperties, Client, ConnectionProperties};
+use lapin_futures as lapin;
+use lapin_futures::ConfirmationFuture;
+use serde::{Deserialize, Serialize};
+use std::{thread, time::Duration};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct RabbitMQSinkConfig {
+  addr: String,
+  encoding: Encoding,
+  queue_name: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum Encoding {
+  Text,
+  Json,
+}
+
+pub struct RabbitMQSink {
+  acker: Acker,
+  channel: lapin_futures::Channel,
+  encoding: Encoding,
+  in_flight: FuturesUnordered<MetadataFuture<ConfirmationFuture<()>, ()>>,
+  queue_name: String,
+}
+
+impl RabbitMQSink {
+  fn new(config: RabbitMQSinkConfig, acker: Acker) -> crate::Result<Self> {
+    let channel = Client::connect(&config.addr, ConnectionProperties::default())
+      .and_then(|client| client.create_channel())
+      .wait()?;
+    channel
+      .queue_declare(
+        &config.queue_name,
+        QueueDeclareOptions::default(),
+        FieldTable::default(),
+      )
+      .wait()?;
+    Ok(RabbitMQSink {
+      acker,
+      channel,
+      encoding: config.encoding,
+      in_flight: FuturesUnordered::new(),
+      queue_name: config.queue_name,
+    })
+  }
+}
+
+#[typetag::serde(name = "rabbitmq")]
+impl SinkConfig for RabbitMQSinkConfig {
+  fn build(&self, acker: Acker) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+    let sink = RabbitMQSink::new(self.clone(), acker)?;
+    let hc = healthcheck(self.clone());
+    Ok((Box::new(sink), hc))
+  }
+
+  fn input_type(&self) -> DataType {
+    DataType::Log
+  }
+}
+
+impl Sink for RabbitMQSink {
+  type SinkItem = Event;
+  type SinkError = ();
+
+  fn start_send(&mut self, item: Self::SinkItem) -> StartSend<Self::SinkItem, Self::SinkError> {
+    let payload = encode_event(&item, &self.encoding);
+    let future = self.channel.basic_publish(
+      "",
+      &self.queue_name,
+      payload,
+      BasicPublishOptions::default(),
+      BasicProperties::default(),
+    );
+    self.in_flight.push(future.join(future::ok(())));
```

Comment:
you probably want to join the ack id here so that you can ack it below in the ready match arm.